### PR TITLE
feat: add TF listener and lookup TF helpers in BaseControllerInterface

### PR DESCRIPTION
**Pull request by bpapaspyros**
## Description

- #169 

This PR solves the issue by replicating the TF listener functions from `ComponentInterface`.

## Review guidelines

Estimated Time of Review: 8 minutes

## Checklist before merging:

- [x] Confirm that the relevant changelog(s) are up-to-date in case of any user-facing changes

### CODE REVIEW

**Review by domire8 — APPROVED** [review]
Is there even a modification compared to modulo component interface?

**Comment by bpapaspyros** [comment]
> Is there even a modification compared to modulo component interface?

no, just `get_node()->` instead of an actual variable holding the node or clock

**Comment by domire8** [comment]
> no, just `get_node()->` instead of an actual variable holding the node or clock

nice, that triggers a problem actually. If someone tried `add_tf_listener()` on construction, this would give a segmentation error because there is no node on construction. Can you verify that and check that the node is actually available?

File: source/modulo_controllers/src/BaseControllerInterface.cpp [hunk 1]
```diff
@@ -539,6 +542,59 @@ void BaseControllerInterface::add_service(
...
+  translators::read_message(result, transform);
+  return result;
+}
+
+geometry_msgs::msg::TransformStamped BaseControllerInterface::lookup_ros_transform(
+    const std::string& frame, const std::string& reference_frame, const tf2::TimePoint& time_point,
+    const tf2::Duration& duration) {
+  if (this->tf_buffer_ == nullptr || this->tf_listener_ == nullptr) {
+    throw modulo_core::exceptions::LookupTransformException(
+        "Failed to lookup transform: To TF buffer / listener configured.");
```
Comment (domire8) [inline source/modulo_controllers/src/BaseControllerInterface.cpp:588]:
```suggestion
        "Failed to lookup transform: No TF buffer / listener configured.");
```

Comment (domire8) [reply source/modulo_controllers/src/BaseControllerInterface.cpp:588]: You can fix that in the Component interface too please :smile:

**Review by domire8 — APPROVED** [review]
Looks good!

File: source/modulo_controllers/src/BaseControllerInterface.cpp (hunk 1 above)
Comment (bpapaspyros) [reply source/modulo_controllers/src/BaseControllerInterface.cpp:588]: ah nice, didn't even notice..

**Pull request merged**